packages/lab/src/preview.ts

```diff
@@ -162,31 +166,28 @@ export class RisePreview extends DocumentWidget<IFrame, INotebookModel> {
     return this._renderOnSave;
   }
 
-  setActiveCellIndex(index: number, reload = true): void {
-    const iframe = this.content.node.querySelector('iframe')!;
-    if (reload) {
-      this._ready = new PromiseDelegate<void>();
-      const setReady = () => {
-        iframe.contentWindow!.removeEventListener('load', setReady);
-        const waitForReveal = setInterval(() => {
-          if (iframe.contentDocument!.querySelector('.reveal')) {
-            clearInterval(waitForReveal);
-            this._ready.resolve();
-          }
-        }, 500);
-      };
+  setActiveCellIndex(index: number, reload = true): Promise<void> {
```

Comment:
To avoid PromiseDelegate

```suggestion
  async setActiveCellIndex(index: number, reload = true): Promise<void> {
```